### ISSUE TITLE
E2E run serial tests in parallel suite 

### ISSUE THREAD

**Issue by aojea** [kubernetes/kubernetes#111944]
GinkgoV1 doesn't support to run Serial and Parallel tests in the same suite, this caused us to set up different jobs for Serial tests.

@onsi implemented this long standing request in V2 , so users can run Serial tests at the same time, Ginkgo will do the magic

> Serial allows the user to mark specs and containers of specs as only eligible to run in serial. Ginkgo will guarantee that these specs never run in parallel with other specs.

https://onsi.github.io/ginkgo/#the-serial-decorator

This can simplify a lot the maintenance of current test-infra jobs and help Conformance users

/sig testing

**Comment by aojea** [comment]
/cc @pohly @chendave @BenTheElder @johnSchnake 

@chendave since you already do the heavylifting for ginkgov2 check if this is something you are interested and feel to assign it to yourself

**Comment by chendave** [comment]
@aojea  thanks Antonio!  pls let me do some investigation first.

**Comment by BenTheElder** [comment]
Since "serial" means "these tests are disruptive and can be the only test running at the time" it is *inherently* slow to run these ... 

We also exclude these in some CI jobs *because of the time it takes to run them* ... 🤔

**Comment by aojea** [comment]
> Since "serial" means "these tests are disruptive and can be the only test running at the time" it is _inherently_ slow to run these ...
> 
> We also exclude these in some CI jobs _because of the time it takes to run them_ ... thinking

oh, yeah, I forget about that, I was thinking in this open issue for Conformance https://github.com/kubernetes/kubernetes/issues/76344

so, these means we need to handle Dirsuptive and Slow too XD , or just do it only for Conformance, last time I've checked there were only 6 Serial tests in Conformance

**Comment by BenTheElder** [comment]
We'll need to be careful about this, since test times will still go up adding serial tests to jobs that were only doing parallel before.

FWIW I think most of the Serial tests have previously been poor candidates for PR jobs and in post-submit it's relatively fine to have two jobs.

**Comment by pohly** [comment]
I think the conclusion is that serial, fast, non-disruptive tests can run in the normal pre-submits as long as they don't add too much runtime - even fast serial tests add up. We need more data before making a decision...

/remove-lifecycle stale

**Comment by aojea** [comment]
I opened this thinking more in the Conformance suite, that right now runs everything in serial and takes a lot of time

cc: @dims for the hydrophone thing

**Comment by pohly** [comment]
So if the goal is to run "all [conformance] tests" in a single invocation which runs serial tests sequentially and parallel ones in parallel, then this makes sense no matter what the timing of tests is.

**Comment by aojea** [comment]
Conformance kind ~120mins ~400 tests and from those only 25 Serial

![image](https://github.com/kubernetes/kubernetes/assets/6450081/ebb961a1-c292-4b66-95b8-d3ce27d612fe)

Kind master parallel ~30 mins ~880 tests

![image](https://github.com/kubernetes/kubernetes/assets/6450081/95eeaae5-060a-4ba9-9fc8-7684e005c167)

**Comment by dims** [comment]
cc @rjsadow 

@aojea both sonobuoy and hydrophone use the conformance image in k/k. So as long as that get fixed up the right way both will inherit the same.

https://github.com/kubernetes/kubernetes/tree/master/test/conformance/image

**Comment by aojea** [comment]
> @aojea both sonobuoy and hydrophone use the conformance image in k/k. So as long as that get fixed up the right way both will inherit the same.

IIRC it is about the ginkgo commands to execute the tests

**Comment by aojea** [comment]
/reopen

@pohly is working hard on this

**Comment by pohly** [comment]
https://github.com/kubernetes/test-infra/pull/35594 cleans up some kind jobs. https://github.com/kubernetes-sigs/kind/pull/4015 is required to make that technically possible.

**Comment by aojea** [comment]
Just for reference and visibility, quoting Patrick

> It's a conservative choice. Running in parallel could increase flakiness. We do run them in parallel in other jobs.

**Comment by pohly** [comment]
That was about jobs which specifically run conformance testing.

I think we could switch some of them over to parallel testing. If our tests flake when run in parallel, then that is something that we should know about and address because we run the tests in parallel in other tests.

**Comment by BenTheElder** [comment]
Conformance jobs should use a single ginkgo node for the forseeable future, but it would be good to have other jobs run both serial and parallel tests in the same job.

That said, I've been skeptical of prioritizing this because in general tests that need to be marked serial have been very slow and poor quality, it's a pretty big red flag when we know the test needs exclusive access to the cluster. Keeping them in their own jobs and discouraging them isn't necessarily such a bad thing ...

**Comment by pohly** [comment]
Agreed that serial tests should be avoided. But having to maintain separate jobs for them just makes it worse in those cases where they cannot be avoided.
